src/macaron/repo_finder/commit_finder.py

```diff
@@ -473,42 +473,73 @@ def _compute_tag_version_similarity(tag_version: str, tag_suffix: str, version_p
     """
     count = len(version_parts)
     # Reduce count for each direct match between version parts and tag version.
-    tag_version_text = tag_version
+    tag_version_text = tag_version.lower()
     for part in version_parts:
+        part = part.lower()
         if part in tag_version_text:
             tag_version_text = tag_version_text.replace(part, "", 1)
             count = count - 1
 
     # Try to reduce the count further based on the tag suffix.
     if tag_suffix:
-        last_part = version_parts[-1]
+        last_part = version_parts[-1].lower()
         # The tag suffix might consist of multiple version parts, e.g. RC1.RELEASE
         suffix_split = split_pattern.split(tag_suffix)
+        # Try to match suffix parts to version.
+        versioned_string_match = False
```

Comment:
I am a bit unclear on the meaning of this `versioned_string_match` variable. Could you please give me some explanation on its role for matching the suffix parts?
Additionally, in this block:
```
                if versioned_string_match:
                    count = count + 1
                    continue
```
Why does a match would increase the count number if the lower the count, the better :thinking:

Reply:
`versioned_string_match` relates to a part of the suffix that is alphanumeric and is assumed to be some kind of versioning scheme, e.g. RC1, M4, etc. A single one of these is allowed to alter the overall score per comparison (if it matches part of the version), hence why the match is stored as a boolean.

The count increases if a match of that kind has already been found. Essentially the same as a non-matching part.

Reply:
Ah I see. So essentially `v0.5.0-RC1` should be treated as good while `v0.5.0-RC1-RC2-RC3` would be treat as bad, and how bad it is depends on the number of alphanumeric parts after the first matched one (i.e RC1 is the first match one, then RC2, RC3 would be treated as bad).

Reply:
Yes, that's it. And with the above two examples you pose, assuming the version we are after is `RC1` then this part of the code helps ensure that the tag with more suffixes is counted as being a worse match.